dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/ORCIDHelper.java

```diff
@@ -105,43 +104,41 @@ public static String getOrcidWorkString(Entry e, Optional<Version> optionalVersi
+            throws IOException, URISyntaxException, InterruptedException {
+        HttpRequest request = HttpRequest.newBuilder().uri(new URI(baseURL + id + "/work")).header(HttpHeaders.CONTENT_TYPE, "application/vnd.orcid+xml").header(HttpHeaders.AUTHORIZATION, "Bearer " + token).POST(ofString(workString)).build();
+        return HttpClient.newBuilder().proxy(ProxySelector.getDefault()).build().send(request,
+                HttpResponse.BodyHandlers.ofString());
     }
 
     /**
-     * This creates a new ORCID work
+     * This updates an existing ORCID work
+     * @return
      */
-    public static CloseableHttpResponse postWorkString(String baseURL, String id, String workString, String token) throws IOException {
-        HttpPost postRequest = new HttpPost(baseURL + id + "/work");
-        postRequest.addHeader(HttpHeaders.CONTENT_TYPE, "application/vnd.orcid+xml");
-        postRequest.addHeader(HttpHeaders.AUTHORIZATION, "Bearer " + token);
-        StringEntity workEntity = new StringEntity(workString);
-        postRequest.setEntity(workEntity);
-        CloseableHttpClient httpClient = HttpClientBuilder.create().build();
-        try {
-            return httpClient.execute(postRequest);
-        } finally {
-            httpClient.close();
-        }
+    public static HttpResponse<String> putWorkString(String baseURL, String id, String workString, String token, String putCode)
+            throws IOException, URISyntaxException, InterruptedException {
+        HttpRequest request = HttpRequest.newBuilder().uri(new URI(baseURL + id + "/work/" + putCode)).header(HttpHeaders.CONTENT_TYPE, "application/vnd.orcid+xml").header(HttpHeaders.AUTHORIZATION, "Bearer " + token).PUT(ofString(workString)).build();
+        return HttpClient.newBuilder().proxy(ProxySelector.getDefault()).build().send(request,
+                HttpResponse.BodyHandlers.ofString());
     }
 
+
     /**
-     * This updates an existing ORCID work
+     * Get the ORCID put code from the response
+     *
+     * @param httpResponse
+     * @return
      */
-    public static CloseableHttpResponse putWorkString(String baseURL, String id, String workString, String token, String putCode) throws IOException {
-        HttpPut postRequest = new HttpPut(baseURL + id + "/work/" + putCode);
-        postRequest.addHeader(HttpHeaders.CONTENT_TYPE, "application/vnd.orcid+xml");
-        postRequest.addHeader(HttpHeaders.AUTHORIZATION, "Bearer " + token);
-        StringEntity workEntity = new StringEntity(workString);
-        postRequest.setEntity(workEntity);
-        try (CloseableHttpClient httpClient = HttpClientBuilder.create().build();) {
-            return httpClient.execute(postRequest);
+    public static String getPutCodeFromLocation(HttpResponse httpResponse) {
+        Optional<String> location = httpResponse.headers().firstValue("Location");
+        URI uri;
+        try {
+            uri = new URI(location.get());
```

Comment:
Is it safe to assume location is never empty?

Reply:
i think it's fairly safe. an unsuccessful response would've thrown an exception before this